### PR TITLE
(In Code Review) Add all opportunity based endpoints

### PR DESCRIPTION
**Pull request by mattkranzler5**
https://jira.desk.technology/browse/APIJ-39

### CODE REVIEW

File: api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+public class Opportunity implements Serializable {
+
+    private static final long serialVersionUID = -6465935225027203799L;
+
+    private int id;
+    private String name;
+    private int probability;
+    private Date createdAt;
+    private Date updatedAt;
+    private BigDecimal amount;
```
Comment (jerrellmardis) [inline api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java:49]: Why `BigDecimal`?

File: api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    private int amountCents;
+    private String amountFormat;
+    private String currencyIsoCode;
+    private String description;
+    private String[] labels;
+    private int[] labelIds;
+    private Date closeDate;
+    private HashMap<String, String> customFields;
```
Comment (jerrellmardis) [inline api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java]: Should use `Map` instead.

File: api-client/src/main/java/com/desk/java/apiclient/model/OpportunityStage.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ */
+public class OpportunityStage implements Serializable {
+
+    private static final long serialVersionUID = 8122722623798786167L;
+
+    private int id;
+    private String name;
+    private int probability;
+    private OpportunityStageType opportunityStageType;
+    private Links _links;
```
Comment (jerrellmardis) [inline api-client/src/main/java/com/desk/java/apiclient/model/OpportunityStage.java]: `_links` should be `links` with the SerializedName of `_links`

**Comment by jerrellmardis** [comment]
First pass done

File: api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java (hunk 1 above)
Comment (mattkranzler5) [inline api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java:49]: For currency amounts it is recommended to use `BigDecimal` in java. See http://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency

Comment (jerrellmardis) [inline api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java]: :+1:

**Pull request merged**